packages/aws-client/src/S3.ts

```diff
@@ -779,39 +706,148 @@ export const uploadS3FileStream = (
       Body: fileStream,
     },
   });
+/**
+ * List of outputs from ListS3Objects functions (including V2)
+ * Guarantees that Key exists
+ */
```

Comment:
I'm not sure it's a good idea to reference `hidden` type _Object

Reply:
how else would you suggest updating this, is an object which overrides it without inheriting it a better idea?

Reply:
in terms of naming, my intention is to name it after its place in *our* s3 module. in this case I've overwritten the outputs of all of our ListS3Object* functions to return this type

Reply:
I prefer the old return type `Promise<ListObjectsV2Output['Contents']>`

Reply:
but that isn't a correct type, as Key is guaranteed *not* undefined. switching back to that just means turning off type checking for lines that ask about Key, or de-optimizing our code to check for Key every time we use it